packages/react/src/command-checkbox-item/CommandCheckboxItem.tsx

```diff
@@ -0,0 +1,69 @@
+import { forwardRef } from "react";
+
+import { Box, type BoxProps } from "../box";
+import { Checkbox } from "../checkbox";
+import { useComboboxContext } from "../combobox-context";
+import { CommandItem } from "../command-item";
+
+type CommandCheckboxItemProps = {
+  onCheckedChange?: (checked: boolean) => void;
+} & Omit<BoxProps<typeof CommandItem>, "onSelect">;
+
+export const CommandCheckboxItem = forwardRef<
+  HTMLDivElement,
+  CommandCheckboxItemProps
+>(({ children, disabled, onCheckedChange, value, ...props }, ref) => {
+  const {
+    mode,
+    setValue,
+    value: contextValue,
+  } = useComboboxContext("ComboboxContext");
+
+  const isChecked = Array.isArray(contextValue)
+    ? contextValue.includes(value)
+    : contextValue === value;
+
+  const handleChange = () => {
+    const newChecked = !isChecked;
+    if (mode === "multiple") {
+      if (Array.isArray(contextValue)) {
+        setValue(
+          newChecked
+            ? [...contextValue, value]
+            : contextValue.filter((v) => v !== value),
+        );
+      } else {
+        setValue([value]);
+      }
```

Comment:
i don't think we need this isArray check here? we can simplify this
```suggestion
      setValue(
        newChecked
          ? [...contextValue, value]
          : contextValue.filter((v) => v !== value),
      );
```